common/coconut-interface/src/lib.rs

```diff
@@ -0,0 +1,291 @@
+pub use coconut_rs::*;
+use getset::{CopyGetters, Getters};
+use serde::{Deserialize, Serialize};
+use url::Url;
+pub use validator_client::validator_api::Client as ValidatorAPIClient;
+use validator_client::validator_api::{
+    error::ValidatorAPIClientError, VALIDATOR_API_BLIND_SIGN, VALIDATOR_API_CACHE_VERSION,
+    VALIDATOR_API_VERIFICATION_KEY,
+};
+
+#[derive(Serialize, Deserialize, Getters, CopyGetters, Clone)]
+pub struct Credential {
+    #[getset(get = "pub")]
+    n_params: u32,
+    #[getset(get = "pub")]
+    theta: Theta,
+    public_attributes: Vec<String>,
+    #[getset(get = "pub")]
+    signature: Signature,
+}
+impl Credential {
+    pub fn new(
+        n_params: u32,
+        theta: &Theta,
+        public_attributes: &[Attribute],
+        signature: &Signature,
+    ) -> Credential {
+        Credential {
+            n_params,
+            theta: theta.clone(),
+            public_attributes: public_attributes
+                .iter()
+                .map(|attr| attr.to_bs58())
+                .collect(),
+            signature: *signature,
+        }
+    }
+
+    pub async fn init(validator_urls: Vec<String>) -> Result<Self, String> {
+        let mut state = State::init();
+        let client = ValidatorAPIClient::new();
+        let signature = get_aggregated_signature(validator_urls.clone(), &state, &client)
```

Comment:
With the said Error type you would be able to just use `?` : )

Reply:
Yeah, I just wanted a more verbose error message :)